library/lua/dfhack.lua

```diff
@@ -97,6 +112,11 @@ local function call_delete(obj)
     if obj then obj:delete() end
 end
 
+---@generic T
+---@param obj DFObject
+---@param fn fun(...):T
+---@param ...? any
+---@return T
```

Comment:
`T` can be multiple values. Is that adequately expressed with a single `T`?

Reply:
Changed to variable return but as stated above LuaLS doesn't acknowledge that with generics. The rest of the values will be unknown.